cpp/tests/test_odesecir.cpp

```diff
@@ -1090,6 +1090,17 @@ TEST(Secir, check_constraints_parameters)
     model.parameters.set<mio::osecir::CriticalPerSevere<double>>(0.5);
     model.parameters.set<mio::osecir::DeathsPerCritical<double>>(1.1);
     ASSERT_EQ(model.parameters.check_constraints(), 1);
+
+    model.parameters.set<mio::osecir::DeathsPerCritical<double>>(0.5);
+    model.parameters.set<mio::osecir::TestAndTraceCapacity<double>>(-1);
```

Comment:
I think you should keep the order of the check_constraints function here and put this lines in between ICUCapacity and TimeExposed. Also for the other tests.
The check at the end to see if false comes out again is very useful, maybe add a similar check for apply_constraints?.